### PR TITLE
Add qtkeychain 0.9.1 support to Windows build.

### LINKED ISSUE #48: Windows: QtKeychain support.

### PR DESCRIPTION
**Pull request by daschuer**
Fixes #48.

### CODE REVIEW

**Comment by daschuer** [comment]
Works, merge? This environment is used for master as well.

File: build_qtkeychain.bat [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+REM TODO: verify that the cmake setting BUILD_SHARED_LIBS works correctly.
+SET ADDITIONAL_SETTINGS=
+IF %STATIC_LIBS% (
+SET ADDITIONAL_SETTINGS=-DQTKEYCHAIN_STATIC=ON
+)
+
+"%CMAKEDIR%\cmake" ../ -G %COMPILERPROJECT% %ADDITIONAL_SETTINGS% -DBUILD_TEST_APPLICATION=OFF -DCMAKE_PREFIX_PATH=%ROOT_DIR%\Qt-%QT_VERSION%\lib\cmake
+IF ERRORLEVEL 1 (
+    SET VALRETURN=1
+	goto END
```
Comment (rryan) [inline build_qtkeychain.bat]: Can you convert tabs to spaces in this file please?

**Comment by rryan** [comment]
> Works, merge? This environment is used for master as well.

LGTM, thanks! Did you test these environments by building Mixxx with it?
http://downloads.mixxx.org/builds/buildserver/2.2.x-windows/prs/2.2-PR53-j00045-x86-release-fastbuild-static-4cc84c5-minimal.zip
http://downloads.mixxx.org/builds/buildserver/2.2.x-windows/prs/2.2-PR53-j00045-x64-release-fastbuild-static-4cc84c5-minimal.zip

**Comment by rryan** [comment]
It looks like the qtkeychain build succeeds but 2 of the 3 files aren't copied for the amd64 build (i386 looks ok).

https://builds.renegadetech.mixxx.org/job/v2.2/job/buildserver-2.2.x-windows-prs/45/architecture=amd64,only_buildserver_capable_nodes=buildserver,platform=windows,release_type=ReleaseFastbuild/consoleText
```
The system cannot find the path specified.
The system cannot find the path specified.
        1 file(s) copied.
```

**Comment by daschuer** [comment]
> Did you test these environments by building Mixxx with it?

Can I do this remote? How?

File: build_qtkeychain.bat [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+)
+
+    
+"%CMAKEDIR%\cmake" .. -G %COMPILERPROJECT% %ADDITIONAL_SETTINGS% -DBUILD_TEST_APPLICATION=OFF -DCMAKE_PREFIX_PATH=%ROOT_DIR%\Qt-%QT_VERSION%\lib\cmake -DCMAKE_INSTALL_PREFIX=%ROOT_DIR% -DCMAKE_INSTALL_INCLUDEDIR=%INCLUDE_DIR%
+IF ERRORLEVEL 1 (
+    SET VALRETURN=1
+    goto END
+)
+
+"%CMAKEDIR%\cmake" --build . --config %CONFIG% --clean-first
```
Comment (rryan) [inline build_qtkeychain.bat:44]: I'm worried about this -- can we stick with how the other cmake builds are done and generate a VS solution, then use %MSBUILD% to build it? The risk is that we might build with the wrong platform toolset, or the wrong runtime type. We use the %MSBUILD% environment variable to control this.

Comment (daschuer) [reply build_qtkeychain.bat:44]: cmake does this behind the scenes and calls msbuild with the right parameter set. It always uses the right msbuild matching the used compiler. IMHO this is more save than worrying with that ourselves. 
This the way Qt Creator uses cmake by the way. I think we can rely on it.

Comment (daschuer) [reply build_qtkeychain.bat:44]: I think we should else consider to change the others as well.

Comment (rryan) [reply build_qtkeychain.bat:44]: Thanks -- I agree now that we resolved my worries in #58. Can you add the `-T %PLATFORM_TOOLSET%` here too?

File: build_environment.bat [hunk 3]
```diff
@@ -311,7 +311,7 @@ SET /A TASKNUM=1
  REM build_chromaprint.bat depends on fftw3
  REM build_taglib.bat depends on zlib
  REM build_qt4.bat depends on sqlite3, zlib, openssl
- REM build_qt5.bat depends on sqlite3, zlib, openssl
+ REM build_qtkeychain.bat depends on qt5
```
Comment (rryan) [inline build_environment.bat]: did you mean to copy this line?

File: build_qtkeychain.bat [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    goto END
+)
+
+"%CMAKEDIR%\cmake" --build . --config %CONFIG% --clean-first 
+IF ERRORLEVEL 1 (
+    SET VALRETURN=1
+    goto END
+)
+
+"%CMAKEDIR%\cmake" --build . --target install
```
Comment (rryan) [inline build_qtkeychain.bat]: I think you need a `--config %CONFIG%` here as in #58.

File: build_qtkeychain.bat [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    goto END
+)
+
+"%CMAKEDIR%\cmake" --build . --target install 
+IF ERRORLEVEL 1 (
+    SET VALRETURN=1
+    goto END
+)
+
+copy qt5keychain.dir\RelWithDebInfo\qt5keychain.pdb %LIB_DIR%
```
Comment (rryan) [inline build_qtkeychain.bat]: replace RelWithDebInfo with %CONFIG% here?

**Comment by rryan** [comment]
All files (.lib, .pdb., headers) seem present to me across both architectures :+1:

**Comment by daschuer** [comment]
Great, so we can merge this?

**Comment by rryan** [comment]
Sounds good! Thanks

**Pull request merged**